src/com/xilinx/rapidwright/edif/AbstractEDIFParser.java

```diff
@@ -0,0 +1,501 @@
+/* 
+ * Copyright (c) 2022 Xilinx, Inc. 
+ * All rights reserved.
+ *
+ * Author: Jakob Wenzel, Xilinx Research Labs.
+ *  
+ * This file is part of RapidWright. 
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ */
+ 
+package com.xilinx.rapidwright.edif;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.xilinx.rapidwright.util.StringPool;
+
+public abstract class AbstractEDIFParser {
```

Comment:
Maybe this could be called `AbstractEDIFParserWorker` because it doesn't provide the base class for both the seq and parallel EDIFParser-s, just the core worker functionality?